service/src/main/kotlin/fi/espoo/evaka/document/archival/ArchiveChildDocumentService.kt

```diff
@@ -467,6 +470,16 @@ fun uploadToArchive(
         throw RuntimeException("Failed to archive document $documentId")
     }
 
+    // Audit log the instance ID
+    if (instanceId == null) {
+        logger.error { "No instance ID found in response body" }
+    } else {
+        Audit.ChildDocumentArchivedSuccessfully.log(
+            targetId = AuditId(documentId),
+            meta = mapOf("instanceId" to instanceId),
+        )
+    }
```

Comment:
Onko jotain syytä miksi tämän pitää mennä just audit lokiin? Jos sitä tarvitaan johonkin bisneslogiikkaan myöhemmin, sen pitäisi mennä tietokantaan. Jos tämä on vain infoa niin se voi yhtä hyvin mennä tavalliseen lokiin.

Audit-lokitusta on evakassa käytetty pääasiassa siihen että pystytään jälkikäteen selvittämään mitä jokin käyttäjä on tehnyt tai mitkä käyttäjät ovat tehneet jonkun operaation. Async jobiin voi assosioida `AuthenticatedUser`-objektin, mutta tässä jobityypissä sitä ei ole tehty, joten lokille ei taida päätyä mitään käyttäjä-id:tä.

Reply:
Minusta nämä Särmän `instanceId`:t täytyy jotenkin säilyttää todisteena siitä, että Särmä on vastaanottanut asiakirjat. Tietokantaan en alkanut tallentamaan näitä, koska ajatus on, että dokumentit ja kaikki muutkin lapsen tiedot poistetaan eVakasta pian asiakirjojen ulkoisen arkistoinnin jälkeen.

Tämä auttaa ongelmatilanteessa tutkimaan, että missä järjestelmässä on vika jos asiakirjojen arkistoinnissa menee jotain pieleen. Normaalin `appMisc` logituksen sijaa valitsin audit logituksen, koska potentiaaliset ongelmat voivat tulla esiin vasta, kun normaalit logit ovat tuhottu.

Olisikohan näille joku parempi paikka kuitenkin kuin audit logi? Vai onkohan koko ajatus `instanceId`iden tallennuksesta turha?

Reply:
No ei mulla kovin vahvaa mielipidettä ole 🤷 Audit lokien pidempi tallennusaika on ihan hyvä pointti, koska arkistointiajat on pitkiä. Mutta jos sinne arkistoon jotain yksittäistä on jäänyt menemättä niin en ole varma lähdetäänkö sitä kovin kalliilla tutkimaan kaivelemalla jotain jäädytettyjä audit lokeja kuitenkaan 🤔